File: previousJava/src/Enumerise/EquationKind.java
package Enumerise;

import DataStruct.CoefP;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import model.AlimIndic;

public enum EquationKind {
	ENERGYNEED("energyNeedDesc", "energyNeed", 0),
	ENERGYDENSITY("energyDensityDesc", "energyDensity", 1),
	MW("metabolicWeightDesc", "metabolicWeight", 2),
	INDICATOR("indicatorDesc", "indicator", 3),
	NEED("NeedDesc", "NeedEq", 4)
	
	;
	
private String description ="";
private String name ="";
private int uuid=0;
EquationKind(String desc, String name, int i){
	this.description=desc;
	this.name=name;
	this.uuid=i;	
}

public String getDescription() {
	return description;
}
public String getName() {
	return name;
}
public int getUuid() {
	return uuid;
}

public static EquationKind getById (int i) {
	
	for (EquationKind ek:EquationKind.values()) {
		if (ek.getUuid()==i) {
			return ek;
		}
	}
	return EquationKind.ENERGYNEED;
}
 
}

File: composeApp/src/commonMain/kotlin/fr/vetbrain/vetnutri_mp/Enumer/EquationKind.kt
package fr.vetbrain.vetnutri_mp.Enumer

enum class EquationKind(
        private val description: String,
        private val nom: String,
        private val uuid: Int
) {
    ENERGYNEED("energyNeedDesc", "energyNeed", 0),
    ENERGYDENSITY("energyDensityDesc", "energyDensity", 1),
    MW("metabolicWeightDesc", "metabolicWeight", 2),
    INDICATOR("indicatorDesc", "indicator", 3),
    NEED("NeedDesc", "NeedEq", 4),
    COMPLEMENTARY_NUTRIENT("complementaryNutrientDesc", "complementaryNutrient", 5),
      ENERCOMP("energyCompDesc", "energyComp", 6);

    fun getDescription() = description
    fun getNom() = nom
    fun getUuid() = uuid

    companion object {
        fun getById(i: Int) = values().find { it.uuid == i } ?: ENERGYNEED
    }
}
